src/server/pfs/server/compactor.go

```diff
@@ -68,11 +96,9 @@ func (d *driver) compact(ctx context.Context, ids []fileset.ID) (*fileset.ID, er
 	})
 }
 
-func (d *driver) compactionWorker() {
-	ctx := context.Background()
-	w := work.NewWorker(d.etcdClient, d.prefix, storageTaskNamespace)
+func (c *compactor) compactionWorker(ctx context.Context) {
```

Comment:
Maybe this should be internal for the compactor, instead of having to run it in a goroutine outside of this abstraction?

Reply:
At some point we should develop a pattern around supervising long running processes.  But for now everything is spawned in either the constructor or the master.

Reply:
I think this should just be a goroutine setup in the compactor constructor, and also pipe the env context through and use it for the task queue.